linker/Tests/Mono.Linker.Tests.Cases.Expectations/Assertions/BaseInAssemblyAttribute.cs

```diff
@@ -0,0 +1,5 @@
+﻿namespace Mono.Linker.Tests.Cases.Expectations.Assertions
+{
+	public abstract class BaseInAssemblyAttribute : BaseExpectedLinkedBehaviorAttribute {
```

Comment:
In an up coming PR I will change the following method in ResultChecker from

```
static bool IsTypeInOtherAssemblyAssertion (CustomAttribute attr)
{
    return attr.AttributeType.Name == nameof (RemovedTypeInAssemblyAttribute)
        || attr.AttributeType.Name == nameof (KeptTypeInAssemblyAttribute)
        || attr.AttributeType.Name == nameof (RemovedMemberInAssemblyAttribute)
        || attr.AttributeType.Name == nameof (KeptMemberInAssemblyAttribute);
}
```
to
```
bool IsTypeInOtherAssemblyAssertion (CustomAttribute attr)
{
    return attr.AttributeType.Resolve ().DerivesFrom (nameof (BaseInAssemblyAttribute));
}
```

Then in our Unity.Linker.Tests.Cases.Expectations project we have an attribute [StubbedMemberInAssembly] that derives from BaseInAssemblyAttribute.  With this change ResultChecker.IsTypeInOtherAssemblyAssertion is able to catch our attribute as well.  My next PR will add a few extensibility points to ResultChecker so that having ResultChecker.IsTypeInOtherAssemblyAssertion catch our attribute is actually useful.